addons/stock/models/stock_package_level.py

```diff
@@ -50,10 +51,11 @@ def _set_is_done(self):
         for package_level in self:
             if package_level.is_done:
                 if not package_level.is_fresh_package:
+                    ml_update_dict = defaultdict(float)
                     for quant in package_level.package_id.quant_ids:
                         corresponding_ml = package_level.move_line_ids.filtered(lambda ml: ml.product_id == quant.product_id and ml.lot_id == quant.lot_id)
                         if corresponding_ml:
-                            corresponding_ml[0].qty_done = corresponding_ml[0].qty_done + quant.quantity
+                            ml_update_dict[corresponding_ml] += quant.quantity
```

Comment:
what if more than one lines are matching?

Reply:
nice catch, I forgot the [0]